functions-and-scope.js

```diff
@@ -57,25 +154,68 @@ const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
 /* 2c: Afronden op twee decimalen */
 // Zorg ervoor dat het gemiddelde cijfer dat wordt teruggegeven uit de functie netjes wordt afgerond op twee decimalen.
 // Tip: Google is your best friend!
-
+//===========================================
+// console.log('Opdr.2b');
+// console.log("/* 2c: Afronden op twee decimalen */");
+// {
+//     console.log(fncAvgGrades(grades));
+//     console.log(fncAvgGrades(grades = [6, 4, 5]));
+//     console.log(fncAvgGrades(grades = [8, 9, 4, 6, 10]));
+//
+//     function fncAvgGrades(arrGrades) {
+//         let intTotGrades = 0;
+//         for (let i = 0; i < arrGrades.length; i++) {
+//             intTotGrades = intTotGrades+grades[i];
+//         }
+//         let dblAvgGrades = (intTotGrades / arrGrades.length);
+//         return dblAvgGrades.toFixed(2);
+//     }
+// }
+// console.log('Eind Opdr.2b\n');
+//======================================================================================================================
 
 
 
 /* Bonusopdracht: hoogste cijfer */
 
 /* 3a: Script schrijven  */
-// Schrijf een script die op basis van de grades array (hierboven) checkt wat het hoogst behaalde cijfer is. Je mag hier geen bestaande methoden voor gebruiken. Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
+// Schrijf een script die op basis van de grades array (hierboven) checkt wat het hoogst behaalde cijfer is. Je mag
+// hier geen bestaande methoden voor gebruiken. Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
 // * Hoe kan ik iedere waarde van de array langsgaan?
 // * Op welke conditie moet ik checken?
 // * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan?
 // Log het antwoord in de terminal.
 
 // ---- Verwachte uitkomst: 9
-
+//===========================================
+// console.log('Opdr.3a+b');
+// console.log("// ---- Verwachte uitkomsten:\n" +
+//     "// highestGrade(grades) geeft 9\n" +
+//     "// highestGrade([6, 4, 5]) geeft 6\n" +
+//     "// highestGrade([8, 9, 4, 6, 10]) geeft 10");
+// {
+//     console.log(highestGrade(grades));
+//     console.log(highestGrade(grades = [6, 4, 5]));
+//     console.log(highestGrade(grades = [8, 9, 4, 6, 10]));
+//
+//     function highestGrade(arrGrades) {
+//         let intHighestGrade = 0;
```

Comment:
Het is beter om te beginnen met de eerste waarde van de array in plaats van `0`. Nu gaat het in dit voorbeeld misschien niet helemaal op wat ik nu ga uitleggen: Stel je voor dat je een array heeft van [-2, -4, -1, -5]. Als je deze nu in deze functie stopt, dan komt er `0` uit, terwijl deze niet eens in de array voorkomt.

```suggestion
//         let intHighestGrade = arrGrades[0];
```

Reply:
Dat is een goeie waar ik niet aan gedacht heb... Dank..